### ISSUE TITLE
cryptsetup fail

### ISSUE THREAD

**Issue by carabus407** [termux/termux-root-packages#134]
hi, when i trying to use cryptsetup in termux, this message appear:
 
CANNOT LINK EXECUTABLE: empty/missing DT_HASH/DT_GNU_HASH in "/data/data/com.termux/files/usr/lib/libdevmapper.so.1.02" (new hash type from the future?)                  
page record for 0x7f8399e010 was not found (block_size=64)

**Comment by Grimler91** [comment]
Hi, what's your output of `termux-info`?

**Comment by carabus407** [comment]
Updatable packages:
All packages up to date
Subscribed repositories:
https://dl.bintray.com/xeffyr/x11-packages-21 x11/main
https://dl.bintray.com/xeffyr/unstable-packages-21 unstable/main
https://dl.bintray.com/grimler/science-packages-21 science/stable
https://dl.bintray.com/grimler/termux-root-packages-21 root/stable
https://dl.bintray.com/grimler/game-packages-21 games/stable
https://termux.net stable/main
https://termux.net stable/main
System information:
Linux localhost 3.18.22+ #1 SMP PREEMPT Wed Jan 23 10:40:30 CST 2019 aarch64 An Android
Termux-packages arch:
aarch64
Android version:
6.0
Device manufacturer:
Xiaomi
Device model:
Redmi Note 4X

**Comment by Grimler91** [comment]
[Libdevmapper requires at least android 6](https://github.com/termux/termux-root-packages/blob/1134eb41802e0b0446c619fcdb049988a76beaef/packages/libdevmapper/build.sh#L13) so my initial guess was that you were on android 5 but you should be fine as you have android 6. I'll see if it's possible to upgrade the package and if that'll help

**Comment by carabus407** [comment]
yes, cryptsetup via termux work fine on android 9  (mi pad 4/havoc os), but on android 6 (redmi note 4x nickel) this issue appear.
